### PR TITLE
FilterHelper.py: Move code from Filter.py

### PR DESCRIPTION
**Pull request by anctartica**
Closes https://github.com/coala/coala/issues/4494

### CODE REVIEW

File: coalib/parsing/filters/Filter.py [hunk 1]
```diff
@@ -1,19 +0,0 @@
...
-
-def filter(filter_function):
-    def filter_section_bears(bears, args):
-        return {section:
-                tuple(bear for bear in bears[section]
-                      if filter_function(bear, args))
-                for section in bears}
-
-    @wraps(filter_function)
-    def filter_wrapper(section_to_bears_dict, args):
```
Comment (sangamcse) [inline coalib/parsing/filters/Filter.py]: Don't we need `filter_wrapper`?

Comment (anctartica) [reply coalib/parsing/filters/Filter.py]: This code has been made inline.

**Comment by Vedarth** [comment]
So you have moved all the filter decorator code to the apply_filter function. This will remove the need for `@filter` which used to share the name with `filter` which was built in Python.

File: coalib/parsing/FilterHelper.py [hunk 2]
```diff
@@ -15,7 +15,20 @@ def apply_filter(filter_name, filter_args, all_bears=None):
         raise InvalidFilterException(filter_name)
     if not filter_args or len(filter_args) == 0:
         return all_bears
-    return available_filters[filter_name](all_bears, filter_args)
+
+    filter_function = available_filters[filter_name]
+
+    def filter_section_bears(bears, args):
```
Comment (Makman2) [inline coalib/parsing/FilterHelper.py]: please move it to module scope and precede it with an underscore: `def _filter_section_bears`
Not necessary for the function to be inside `apply_filter` scope

**Comment by Makman2** [comment]
Optional:
Commit message: Maybe `Move` --> `Inline`? Up to you :)

**Comment by Makman2** [comment]
unack 2b8371aa07efe168b636320acf516750cfa5af0a

File: coalib/parsing/FilterHelper.py (hunk 2 above)
Comment (anctartica) [reply coalib/parsing/FilterHelper.py]: okay 👍

**Comment by anctartica** [comment]
@Makman2 made the changes. Please review again

**Comment by Makman2** [comment]
ack 7f9e5be00ef4599e1c4508b05a46cecd52450325

**Comment by Makman2** [comment]
@gitmate-bot rebase

**Comment by Makman2** [comment]
@gitmate-bot ff

**Pull request merged**